core-web/libs/sdk/experiments/src/lib/shared/utils/utils.ts

```diff
@@ -202,3 +189,5 @@ export const objectsAreEqual = (obj1: Experiment[], obj2: Experiment[]): boolean
 
     return JSON.stringify(obj1) === JSON.stringify(obj2);
 };
+
+export const defaultRedirectFn = (href: string) => (window.location.href = href);
```

Comment:
I'm not sure if this is safe

Reply:
I created this one for standalone approach (consume the class as IIFE), but with Next, I pass the replace to redirect in the config to the provider 
``` 
const { refresh, replace, route, asPath, query } = useRouter();

const experimentConfig = {
    'api-key': process.env.NEXT_PUBLIC_EXPERIMENTS_API_KEY ,
    server: process.env.NEXT_PUBLIC_DOTCMS_HOST ,
    debug: process.env.NEXT_PUBLIC_EXPERIMENTS_DEBUG,
    redirectFn: replace
  }

```

and sent it to the class with
```
 instance.locationChanged(location, redirectFn);
```
and inside the method I do this:
```
 if (redirectFunction) {
                    this.logger.log(
                        `Page redirected to ${variantUrl} using the provided redirect function.`
                    );

                    redirectFunction(variantUrl);
                } else {
                    this.logger.log(
                        `Page redirected to ${variantUrl} using the default redirect function.`
                    );

                    defaultRedirectFn(variantUrl);
                }
```

This allow to use the redirect of the framework/metaframework/lib already has to do the redirect.

Reply:
Mmm... It looks good